Core/GDCore/Project/Project.cpp

```diff
@@ -102,29 +102,17 @@ std::unique_ptr<gd::Object> Project::CreateObject(
     behavior->SetDefaultBehavior(true);
   };
 
-  if (project.HasEventsBasedObject(objectType)) {
-    // During project deserialization, event-based object metadata are not yet
-    // generated.
-    addDefaultBehavior("EffectCapability::EffectBehavior");
-    addDefaultBehavior("ResizableCapability::ResizableBehavior");
-    addDefaultBehavior("ScalableCapability::ScalableBehavior");
-    addDefaultBehavior("FlippableCapability::FlippableBehavior");
-    auto& eventBasedObject = project.GetEventsBasedObject(objectType);
-    if (eventBasedObject.IsRenderedIn3D()) {
-      addDefaultBehavior("Scene3D::Base3DBehavior");
-    }
-    else {
-      addDefaultBehavior("OpacityCapability::OpacityBehavior");
-    }
-  } else {
-    auto& objectMetadata =
-        gd::MetadataProvider::GetObjectMetadata(platform, objectType);
-    if (MetadataProvider::IsBadObjectMetadata(objectMetadata)) {
-      gd::LogWarning("Object: " + name + " has an unknown type: " + objectType);
-    }
-    for (auto& behaviorType : objectMetadata.GetDefaultBehaviors()) {
-      addDefaultBehavior(behaviorType);
-    }
+  auto &objectMetadata =
+      gd::MetadataProvider::GetObjectMetadata(platform, objectType);
+  if (MetadataProvider::IsBadObjectMetadata(objectMetadata)
+      // During project deserialization, event-based object metadata are not yet
+      // generated. Default behaviors will be added by
+      // MetadataDeclarationHelper::UpdateCustomObjectDefaultBehaviors
+      && !project.HasEventsBasedObject(objectType)) {
+    gd::LogWarning("Object: " + name + " has an unknown type: " + objectType);
+  }
+  for (auto &behaviorType : objectMetadata.GetDefaultBehaviors()) {

```

Comment:
So if the object metadata is "badmetadata", because it's an events based object, we don't add default behaviors there (the loop will loop on nothing). Is:
- what I say right? 
- could we avoid doing this at all, even for non events based objects?

Reply:
Yes, do you mean it should return before the empty loop? Isn't the purpose of bad metadata to avoid to care?

Reply:
It's more like "avoiding to crash by doing a noop in case you forget it" :)

Let's add a comment `// This is a no-op in case the metadata of an object is not found - which can happen the first time a custom object is loaded.`